build-system/tasks/clean.js

```diff
@@ -15,28 +15,51 @@
  */
 'use strict';
 
+const argv = require('minimist')(process.argv.slice(2));
 const del = require('del');
+const log = require('fancy-log');
+const path = require('path');
+const {cyan} = require('ansi-colors');
+
+const ROOT_DIR = path.resolve(__dirname, '../../');
 
 /**
- * Clean up the build artifacts
- * @return {!Promise}
+ * Cleans up various build and test artifacts
  */
 async function clean() {
-  return del([
-    'dist',
-    'dist.3p',
-    'dist.tools',
-    'build',
+  const pathsToDelete = [
     '.amp-build',
     '.karma-cache',
-    'deps.txt',
+    'build',
     'build-system/server/new-server/transforms/dist',
+    'deps.txt',
+    'dist',
+    'dist.3p',
+    'dist.tools',
     'test-bin',
-  ]);
+  ];
+  if (argv.include_subpackages) {
+    pathsToDelete.push('**/node_modules', '!node_modules');
+  }
+  const deletedPaths = await del(pathsToDelete, {
+    expandDirectories: true,
+    dryRun: argv.dry_run,
+  });
+  if (deletedPaths.length > 0) {
+    log(argv.dry_run ? 'Paths that will be deleted:' : 'Deleted paths:');
```

Comment:
nit: "would be" makes more sense than "will be"

Reply:
You are correct! Changed to past conditional tense.